numpy/ma/extras.py

```diff
@@ -981,7 +981,7 @@ def mask_rows(a, axis=None):
     """
     return mask_rowcols(a, 0)
 
-def mask_cols(a, axis=None):
```

Comment:
The alternative would be to add something like:
```
if axis in (None, 1, -(a.ndim-1)):
    deprecate()
else:
    raise TypeError
```
But this doesn't seem worth it.

Reply:
Well, you can just change to `np._NoValue` and `if axis is not np._NoValue: deprecate()`?

Reply:
This is skipping the deprecation warning for cases when the code is actively wrong - when the axis being used is different to the one the user requeste

Reply:
Is your plan to delete `axis` or to change its meaning in the future?

Reply:
Delete it